devdoc/docs/ContributeDocs.md

```diff
@@ -0,0 +1,394 @@
+
+    -   **Error Handling:** Explain what happens if users encounter
+        errors and how to troubleshoot common issues.
+
+4.  **Limitations and Considerations:**
+
+    -   **Limitations:** Acknowledge any restrictions or boundaries
+        associated with the feature's functionality.
+
+### Optional Components:
+
+1.  **Advanced Usage**
+
+    -   **Detailed configuration options:** If the feature offers
+        advanced settings or customizations, provide in-depth
+        instructions for experienced users.
+
+2.  **API Reference (for technical audiences)**
+
+    -   **Technical specifications:** For features with an API
+        component, include detailed API reference documentation for
+        developers integrating it into their applications.
+
+3.  **FAQs**
+
+    -   **Frequently Asked Questions:** Address any commonly anticipated
+        user questions about the feature to pre-empt confusion.
+
+4.  **Additional Resources**
+
+    -   **Links to related documentation:** Include links to relevant
+        documentation for features that interact with this new addition.
+
+    -   **Tutorials:** Consider creating tutorials for a more
+        interactive learning experience.
+
+    -   **Videos:** Consider requesting that a video be made to provide
+        a more interactive visual learning experience. You should
+        provide a simple script or outline of what should be shown in
+        the video.
+
+### General Tips
+
+-   **Target your audience:** Tailor the level of detail and technical
+    jargon you use based on whether the documentation is for developers
+    or end-users.
+
+-   **Clarity and Conciseness:** Use clear, concise language and
+    maintain a consistent structure for easy navigation. Always use
+    present tense, active voice. Remember, you’re writing for the
+    open-source community, not academics, so keep it simple and
+    straight-forward. See the HPCC Style Guide for additional guidance.
+
+-   **Visual Aids:** Screenshots, diagrams, and flowcharts can
+    significantly enhance understanding. A picture can communicate
+    instantly what a thousand words cannot.
+
+-   **Maintain and Update:** Regularly review and update documentation
+    as the feature evolves or based on user feedback.
+
+By following these guidelines and including the required and optional
+components, you can create comprehensive documentation that empowers
+users and streamlines the adoption of your new software feature.
+
+### Placement
+
+In general, it makes sense to keep your documentation near the code. For
+example, a document about ECL Agent should go in the ECLAgent folder.
+
+However, there are times where that is either not possible or a document
+may cover more than one component. In those cases, there are a few
+options as shown below.
+
+#### Other Folders
+
+**DevDocs**: This is a general folder for any developer document.
+
+**Docs**: This is a folder for documents about documentation.
```

Comment:
Is this meant to be `docs`?  This document about documentation is in `devdoc/docs`

Reply:
yes, but i am open to suggestion for a better name for this folder